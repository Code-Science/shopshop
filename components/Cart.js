import React, { useState, useEffect } from 'react';
import classNames from 'classnames';
import Link from 'next/link';
import PropTypes from 'prop-types';
import CloseIcon from '@material-ui/icons/Close';
import CartItem from './CartItem';
import styles from '../styles/components/Cart.module.css';
import { useStateValue } from '../store/StateProvider';

function Cart({ show, close }) {
  const [cartDisplay, setDisplay] = useState(false);
  const cartClass = classNames({
    [styles.cart]: true,
    [styles.show]: show,
    [styles.cartDisplay]: cartDisplay,
  });

  useEffect(() => {
    const timeout = setTimeout(() => {
      setDisplay(true);
      return () => clearTimeout(timeout);
    }, 500);
  }, []);

  const [{ basket }] = useStateValue();

  const totalPrice = () => {
    return basket.reduce((sum, curr) => {
      return sum + curr.price * curr.quantity;
    }, 0);
  };

  const items = basket.map((item) => {
    return (
      <CartItem
        key={item.id}
        id={item.id}
        title={item.title}
        image={item.image}
        price={item.price}
        quantity={item.quantity}
      />
    );
  });

  return (
    <div className={cartClass}>
      <div className={styles.top}>
        <p>Your Cart</p>
        <button type="button" onClick={close} className={styles.cartBtn}>
          <CloseIcon style={{ fontSize: 30 }} />
        </button>
      </div>
      <div className={styles.selection}>
        <ul className={styles.cartList}>
          {items.length === 0 ? <p>Your cart is empty</p> : items}
        </ul>
      </div>
      <div className={styles.endSection}>
        <p>Special instructions for seller</p>
        <textarea className={styles.textarea}></textarea>
        <p>
          <span>Subtotal : </span>
          <strong className={styles.totalPrice}>
            Â£{totalPrice().toFixed(2)}
          </strong>
        </p>
      </div>
      <div>
        {basket.length > 0 ? (
          <Link href="/checkout">
            <a className={styles.checkoutBtn}>Check Out</a>
          </Link>
        ) : (
          <button type="button" disabled={true} className={styles.checkoutBtn}>
            Check Out
          </button>
        )}
      </div>
    </div>
  );
}

Cart.propTypes = {
  show: PropTypes.bool.isRequired,
  close: PropTypes.func,
};

export default Cart;
