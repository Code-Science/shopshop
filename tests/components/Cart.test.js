import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Cart from '../../components/Cart';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('Cart Component', () => {
  const wrapper = shallow(<Cart show={true} />);

  it('expect to render Cart component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
