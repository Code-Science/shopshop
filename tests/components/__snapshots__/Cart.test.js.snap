// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component snapshot test 1`] = `
<div
  className="cart show"
>
  <div
    className="top"
  >
    <p>
      Your Cart
    </p>
    <button
      className="cartBtn"
      type="button"
    >
      <CloseIcon
        style={
          Object {
            "fontSize": 30,
          }
        }
      />
    </button>
  </div>
  <div
    className="selection"
  >
    <ul
      className="cartList"
    >
      <CartItem
        id={4}
        key="4"
        price={80}
      />
    </ul>
  </div>
  <div
    className="endSection"
  >
    <p>
      Special instructions for seller
    </p>
    <textarea
      className="textarea"
    />
    <p>
      <span>
        Subtotal : 
      </span>
      <strong
        className="totalPrice"
      >
        Â£
        NaN
      </strong>
    </p>
  </div>
  <div>
    <Link
      href="/checkout"
    >
      <a
        className="checkoutBtn"
      >
        Check Out
      </a>
    </Link>
  </div>
</div>
`;
